lib/cache.rb

```diff
@@ -74,17 +74,16 @@ def delete(name)
 
   def fetch(name, expires_in: nil, force: nil, &blk)
     if !block_given?
-      if force
-        raise ArgumentError,
-              "Missing block: Calling `Cache#fetch` with `force: true` requires a block."
-      end
+      raise ArgumentError, "Calling `Cache#fetch` with `force: true` requires a block." if force
       return read(name)
     end
 
     key = normalize_key(name)
-    raw = redis.get(key) if !force
-    entry = read_entry(key) if raw
-    return entry if raw && !(entry == :__corrupt_cache__)
+
+    if !force
```

Comment:
This will change how the cache behaves in a subtle way: if for whatever reason we cached `nil`, this will recompute the value each time we try to access it. I remember when I initially patched this code, I tried to rely only on `read_entry`, but in doing so, it was breaking the expected behavior :confused: 

Are we sure the root cause is a race condition happening in the cache and not something that causes caching `nil` instead of a proper value? (I’m missing some context, sorry if my question is a bit dumb :sweat_smile:)